### PR TITLE
add ghcr.io docker image registry

### PR DESCRIPTION
**Pull request by KingPin**
Docker hub artificially limits pull requests for all users who aren't paid. https://docs.docker.com/docker-hub/download-rate-limit
 (100 for not logged-in users across *all* images pulls for the IP)

And it never hurts to have a second point of images. Especially considering you are already storing your code on GitHub, so having the images here is helpful. 

This PR will use your existing credentials to push an image to docker hub + GitHub packages and will list it on your main code page for https://github.com/msgbyte/tianji. No extra work should be needed.

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ on:
 jobs:
   docker:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
```
Comment (moonrailgun) [inline .github/workflows/docker-publish.yml:14]: is this permission is required?

File: .github/workflows/docker-publish.yml
Comment (moonrailgun) [inline .github/workflows/docker-publish.yml:1]: can you add it also in `docker-publish-canary.yml` so i can test it

**Review by moonrailgun — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (KingPin) [reply .github/workflows/docker-publish.yml:14]: yes in order to write/upload the package (image)

File: .github/workflows/docker-publish.yml
Comment (KingPin) [reply .github/workflows/docker-publish.yml:1]: Of course, I'm working atm, but soon as I have a few minutes I can make a second PR, would that work?

Comment (moonrailgun) [reply .github/workflows/docker-publish.yml:1]: better to do it together... because i need test it.

Comment (KingPin) [reply .github/workflows/docker-publish.yml:1]: ah no issues I can do it in same PR,

Comment (KingPin) [reply .github/workflows/docker-publish.yml:1]: ok done.

**Pull request merged**